projects/net-snmp/parse_octet_hint_fuzzer.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/net-snmp-includes.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+int LLVMFuzzerInitialize(int *argc, char ***argv) {
+    if (getenv("NETSNMP_DEBUGGING") != NULL) {
+        /*
+         * Turn on all debugging, to help understand what
+         * bits of the parser are running.
+         */
+        snmp_enable_stderrlog();
+        snmp_set_do_debugging(1);
+        debug_register_tokens("");
+    }
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    char *hint, *value;
+    size_t value_start, new_val_len;
+    unsigned char *new_val;
+
+    hint = strndup(data, size);
+    value_start = min(strlen(hint) + 1, size);
```

Comment:
This currently blocks compilation (This is why the CI fails):
```
...
/work/parse_octet_hint_fuzzer.o: In function `LLVMFuzzerTestOneInput':
/src/parse_octet_hint_fuzzer.c:42: undefined reference to `min'
/src/parse_octet_hint_fuzzer.c:42: undefined reference to `min'
```

Reply:
Right, min() is available in the Linux kernel but not in ANSI C. I realized that it is not necessary to use min() here.